agent/config/types.go

```diff
@@ -101,8 +101,8 @@ type Config struct {
 	// on the instance
 	DisableDockerHealthCheck BooleanDefaultFalse
 
-	// ReservedMemory specifies the amount of memory (in MB) to reserve for things
-	// other than containers managed by ECS
+	// ReservedMemory specifies the amount of memory (in MB) reduction for things
+	// other than containers managed by ECS when registering instance
```

Comment:
Assuming that `memResource` is honored when managing task containers, the code below tells me that `ReservedMemory` is indeed the amount of memory we do not use for ECS containers. Why is the current description inaccurate?
```go
        cpu, mem := getCpuAndMemory()
	remainingMem := mem - int64(client.config.ReservedMemory)
	seelog.Infof("Remaining mem: %d", remainingMem)
	if remainingMem < 0 {
		return nil, fmt.Errorf(
			"api register-container-instance: reserved memory is higher than available memory on the host, total memory: %d, reserved: %d",
			mem, client.config.ReservedMemory)
	}

	cpuResource := ecs.Resource{
		Name:         utils.Strptr("CPU"),
		Type:         &integerStr,
		IntegerValue: &cpu,
	}
	memResource := ecs.Resource{
		Name:         utils.Strptr("MEMORY"),
		Type:         &integerStr,
		IntegerValue: &remainingMem,
	}
```
https://github.com/aws/amazon-ecs-agent/blob/master/agent/api/ecsclient/client.go#L272

Reply:
Yes, this is not used by containers, because when we register instance with Control Plane, the amount of memory is the result of this deduction. The problem brought up in that ticket is - this gives customer a false expectation that any excessive usage of resources beyond this limited is not allowed, in other word, this is treated as a hard limit.

Reply:
I see what you mean now.